internal/auth/oauth.go

```diff
@@ -0,0 +1,179 @@
+		}
+
+		return []string{"user:email", "repo", "read:packages", "write:packages", "workflow", "read:org"}
+	}
+
+	endpoint := func(provider string) oauth2.Endpoint {
+		if provider == Google {
+			return google.Endpoint
+		}
+
+		return github.Endpoint
+	}
+
+	if provider != Google && provider != Github {
+		return nil, newReadError("provider", provider, err)
+	}
+
+	clientId, err := readFileOrConfig(fmt.Sprintf("%s.clientID", provider)) //nolint:stylecheck
+
+	if err != nil {
+		return nil, newReadError("clientID", provider, err)
+	}
+
+	clientSecret, err := readFileOrConfig(fmt.Sprintf("%s.client_secret", provider))
+	if err != nil {
+		return nil, newReadError("clientID", provider, err)
+	}
+
+	return &oauth2.Config{
+		ClientID:     clientId,
+		ClientSecret: clientSecret,
+		RedirectURL:  redirectURL(provider, cli),
+		Scopes:       scopes(provider),
+		Endpoint:     endpoint(provider),
+	}, nil
+}
+
+// readFileOrConfig prefers reading from configKey_file for mounting of a file when running in an environment like Kubernetes, but falls back to a viper string value if the file is not present
+func readFileOrConfig(configKey string) (string, error) {
+	fileKey := configKey + "_file"
+	if viper.IsSet(fileKey) && viper.GetString(fileKey) != "" {
+		filename := viper.GetString(fileKey)
+		// filepath.Clean avoids a gosec warning on reading a file by name
+		data, err := os.ReadFile(filepath.Clean(filename))
+		if err != nil {
+			return "", err
+		}
+
+		return string(data), nil
+	}
+
+	return viper.GetString(configKey), nil
+}
+
+// NewProviderHTTPClient creates a new http client for the given provider
+func NewProviderHTTPClient(provider string) *http.Client {
+	if provider == Github {
+		hClient := &http.Client{
+			Transport: &go_github.BasicAuthTransport{
+				Username: viper.GetString(fmt.Sprintf("%s.clientID", provider)),
+				Password: viper.GetString(fmt.Sprintf("%s.client_secret", provider)),
+			},
+		}
+
+		return hClient
+	}
+
+	return nil
+}
+
+// DeleteAccessToken deletes the access token for a given provider
+func DeleteAccessToken(ctx context.Context, provider string, token string) error {
+	var err error
+
+	hClient := NewProviderHTTPClient(provider)
+	if hClient == nil {
+		return newReadError("provider", provider, err)
+	}
+
+	client := go_github.NewClient(hClient)
+	clientID := viper.GetString(fmt.Sprintf("%s.clientID", provider))
+	_, err = client.Authorizations.Revoke(ctx, clientID, token)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ValidateProviderToken validates the given token for the given provider
+func ValidateProviderToken(_ context.Context, provider string, token string) error {
+	var err error
+
+	if provider == Github {
+		// Create an OAuth2 token source with the PAT
+		tokenSource := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token})
+
+		// Create an authenticated GitHub client
+		oauth2Client := oauth2.NewClient(context.Background(), tokenSource)
```

Comment:
Do you want to pass that context in and use it? 

```suggestion
func ValidateProviderToken(ctx context.Context, provider string, token string) error {
	var err error

	if provider == Github {
		// Create an OAuth2 token source with the PAT
		tokenSource := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token})

		// Create an authenticated GitHub client
		oauth2Client := oauth2.NewClient(ctx, tokenSource)
```